### PR TITLE
cyprus sweep

### PR DESCRIPTION
**Pull request by thedavidmeister**
- [x] I have added a summary of my changes to the changelog

## changes

- use nodejs 8.x LTS in nixos (well, technically it is whatever nix pkgs provides for 8.x)
- tag and push develop docker box from circle (so that the default is a nix box)
- reinstate the cold os x builds so we know if the mac install script works (because it doesn't)
- delete oooold docker files that are confusing
- use a nixos friendly shebang for bash scripts
- remove the docker/entry file as it only works for ubuntu and breaks nix
- tweak docker run files to fit nix better
- fixes for what is installed by the mac os x install script

## manual steps after merge

**disable the relevant docker hub builds/tags! if you don't know how, please ask me!**

## followup

- only run mac cold builds on develop branch #1028

## notes

ominous foreshadowing:

```
qt is keg-only, which means it was not symlinked into /usr/local,
because Qt 5 has CMake issues when linked.
```

### CODE REVIEW

**Comment by zippy** [comment]
hey @thedavidmeister is this a branch we should be merging in?

**Comment by thedavidmeister** [comment]
@zippy yes it is!

i'll clean it up now

File: shell.nix [hunk 1]
```diff
@@ -99,7 +90,7 @@ stdenv.mkDerivation rec {
     pkgconfig
     rust-build
 
-    nodejs-8_13
+    nodejs-8_x
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply shell.nix]: this reverts to the default upstream from nix, so it is no longer locked to anything other than 8.x

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Changed
+- Node JS version relaxed to 8.x in nix shell
+- develop docker tag now uses nix
```
Comment (zippy) [inline CHANGELOG.md]: how about we use the "removed" section for this line... :-)

File: scripts/install/osx.sh [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
+brew install cmake
 brew install zmq
 brew upgrade zeromq
+brew install qt5
+# https://superuser.com/questions/256232/how-can-i-get-qmake-on-mac-os-x#comment1880535_422785
+# https://superuser.com/a/1153338
+brew link qt5 --force
 
-export INSTALL_NODE_VERSION=8.4
+export INSTALL_NODE_VERSION=8.14
```
Comment (zippy) [inline scripts/install/osx.sh]: is this supposed to be 8.14.0?

**Pull request merged**